File: TreeViewAdvDemo/VB/TreeViewSearchForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports Syncfusion.Windows.Forms.Tools
Imports Syncfusion.Windows.Forms

Namespace TreeViewAdvSearchDemo

    Partial Public Class TreeViewSearchForm
        Inherits MetroForm
#Region "Variables"

        'Initialize the TreeViewAdvFindReplaceDialog
        Private dialog As TreeViewAdvFindReplaceDialog = Nothing
        'Initialize the TreeViewAdv
        Private TreeView As TreeViewAdv
#End Region

#Region "Consructor"

        'To pass the TreeviewAdv argument to the form.
        Public Sub New(ByVal tree As TreeViewAdv)
            InitializeComponent()
            TreeView = tree
            Me.TreeView.InactiveSelectedNodeBackground = New Syncfusion.Drawing.BrushInfo(System.Drawing.Color.FromArgb((CInt(Fix((CByte(17))))), (CInt(Fix((CByte(158))))), (CInt(Fix((CByte(218)))))))
            'To assign the TreeViewAdv to TreeViewSearchForm            
            dialog = New TreeViewAdvFindReplaceDialog(Me.TreeView)

            'To add the TreeViewSearchRange in ComboBox
            Dim searchRange As Array = System.Enum.GetValues(GetType(TreeViewSearchRange))
            For i As Integer = 0 To searchRange.Length - 1
                Me.SearchRangeCombo.Items.Add(searchRange.GetValue(i))
            Next i

            'To add the TreeViewSearchOption in ComboBox
            Dim SearchOptions As Array = System.Enum.GetValues(GetType(TreeViewSearchOption))
            For i As Integer = 0 To SearchOptions.Length - 1
                Me.SearchOptionCombo.Items.Add(SearchOptions.GetValue(i))
            Next i

            'To add the TreeViewSearchNavigation in ComboBox
            Dim NodeSearchType As Array = System.Enum.GetValues(GetType(TreeViewSearchNavigation))
            For i As Integer = 0 To NodeSearchType.Length - 1
                Me.comboBox3.Items.Add(NodeSearchType.GetValue(i))
            Next i
            AddHandler comboBox3.SelectedIndexChanged, AddressOf comboBox3_SelectedIndexChanged
            Me.StartPosition = FormStartPosition.CenterScreen
            Me.SearchOptionCombo.SelectedIndex = 0
            Me.SearchRangeCombo.SelectedIndex = 0
            Me.comboBox3.SelectedIndex = 1
            Me.findTextBox.Select()
        End Sub
#End Region

#Region "Implementation"

        'To get the TreeViewSearchNavigation
        Private Sub comboBox3_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
            dialog.TreeViewSearchNavigation = CType(Me.SearchRangeCombo.Items(Me.comboBox3.Items.IndexOf(Me.comboBox3.SelectedItem)), TreeViewSearchNavigation)
        End Sub

        'To get the TreeViewSearchRange
        Private Sub SearchRangeCombo_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles SearchRangeCombo.SelectedIndexChanged
            dialog.TreeViewSearchRange = CType(Me.SearchRangeCombo.Items(Me.SearchRangeCombo.Items.IndexOf(Me.SearchRangeCombo.SelectedItem)), TreeViewSearchRange)
        End Sub

        'To get the TreeViewSearchOption
        Private Sub SearchOptionCombo_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles SearchOptionCombo.SelectedIndexChanged
            dialog.TreeViewSearchOption = CType(Me.SearchOptionCombo.Items(Me.SearchOptionCombo.Items.IndexOf(Me.SearchOptionCombo.SelectedItem)), TreeViewSearchOption)
        End Sub

        Private Sub button5_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button5.Click
            Me.Close()
        End Sub

        'To Find TreeNodeAdv
        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            dialog.Find(Me.findTextBox.Text)
        End Sub

        'To Find All the TreeNodeAdv
        Private Sub button2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button2.Click
            dialog.FindAll(Me.findTextBox.Text)
        End Sub

        'To Replace the TreeNodeAdv
        Private Sub button3_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button3.Click
            Me.TreeView.SelectedNodes.Clear()
            If dialog.Find(Me.findTextBox.Text) Then
                dialog.Replace(Me.replaceTextbox.Text)
            End If
        End Sub

        'To Replace All the TreeNodeAdv
        Private Sub button4_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button4.Click
            Me.TreeView.SelectedNodes.Clear()
            If dialog.FindAll(Me.findTextBox.Text) Then
                dialog.ReplaceAll(Me.replaceTextbox.Text)
            End If
        End Sub
#End Region
    End Class
End Namespace

File: TreeViewAdvDemo/C#/TreeViewSearchForm.cs
using System;
using System.Windows.Forms;
using Syncfusion.Windows.Forms.Tools;
using Syncfusion.Windows.Forms;

namespace TreeViewAdvSearchDemo
{
    public partial class TreeViewSearchForm : MetroForm
    {
        #region Variables

        //Initialize the TreeViewAdvFindReplaceDialog
        TreeViewAdvFindReplaceDialog dialog = null;
        //Initialize the TreeViewAdv
        private TreeViewAdv TreeView;
        #endregion

        #region Consructor

        //To pass the TreeviewAdv argument to the form.
        public TreeViewSearchForm(TreeViewAdv tree)
        {
            InitializeComponent();
            TreeView = tree;
            this.TreeView.InactiveSelectedNodeBackground = new Syncfusion.Drawing.BrushInfo(System.Drawing.Color.FromArgb(((int)(((byte)(17)))), ((int)(((byte)(158)))), ((int)(((byte)(218))))));
            //To assign the TreeViewAdv to TreeViewSearchForm            
            dialog = new TreeViewAdvFindReplaceDialog(this.TreeView);

            //To add the TreeViewSearchRange in ComboBox
            Array searchRange = Enum.GetValues(typeof(TreeViewSearchRange));
            for (int i = 0; i < searchRange.Length; i++)
            {
                this.SearchRangeCombo.Items.Add(searchRange.GetValue(i));
            }

            //To add the TreeViewSearchOption in ComboBox
            Array SearchOptions = Enum.GetValues(typeof(TreeViewSearchOption));
            for (int i = 0; i < SearchOptions.Length; i++)
            {
                this.SearchOptionCombo.Items.Add(SearchOptions.GetValue(i));
            }

            //To add the TreeViewSearchNavigation in ComboBox
            Array NodeSearchType = Enum.GetValues(typeof(TreeViewSearchNavigation));
            for (int i = 0; i < NodeSearchType.Length; i++)
            {
                this.comboBox3.Items.Add(NodeSearchType.GetValue(i));
            }
            this.comboBox3.SelectedIndexChanged += new EventHandler(comboBox3_SelectedIndexChanged);
            this.StartPosition = FormStartPosition.CenterScreen;
            this.SearchOptionCombo.SelectedIndex = 0;
            this.SearchRangeCombo.SelectedIndex = 0;
            this.comboBox3.SelectedIndex = 1;
            this.findTextBox.Select();
        }
        #endregion

        #region Implementation

        //To get the TreeViewSearchNavigation
        void comboBox3_SelectedIndexChanged(object sender, EventArgs e)
        {
            dialog.TreeViewSearchNavigation = (TreeViewSearchNavigation)this.SearchRangeCombo.Items[this.comboBox3.Items.IndexOf(this.comboBox3.SelectedItem)];
        }

        //To get the TreeViewSearchRange
        private void SearchRangeCombo_SelectedIndexChanged(object sender, EventArgs e)
        {
            dialog.TreeViewSearchRange = (TreeViewSearchRange)this.SearchRangeCombo.Items[this.SearchRangeCombo.Items.IndexOf(this.SearchRangeCombo.SelectedItem)];
        }

        //To get the TreeViewSearchOption
        private void SearchOptionCombo_SelectedIndexChanged(object sender, EventArgs e)
        {
            dialog.TreeViewSearchOption = (TreeViewSearchOption)this.SearchOptionCombo.Items[this.SearchOptionCombo.Items.IndexOf(this.SearchOptionCombo.SelectedItem)];
        }

        private void button5_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        //To Find TreeNodeAdv
        private void button1_Click(object sender, EventArgs e)
        {
            dialog.Find(this.findTextBox.Text);
        }

        //To Find All the TreeNodeAdv
        private void button2_Click(object sender, EventArgs e)
        {
            dialog.FindAll(this.findTextBox.Text);
        }

        //To Replace the TreeNodeAdv
        private void button3_Click(object sender, EventArgs e)
        {
            this.TreeView.SelectedNodes.Clear();
            if (dialog.Find(this.findTextBox.Text))
                dialog.Replace(this.replaceTextbox.Text);
        }

        //To Replace All the TreeNodeAdv
        private void button4_Click(object sender, EventArgs e)
        {
            this.TreeView.SelectedNodes.Clear();
            if (dialog.FindAll(this.findTextBox.Text))
                dialog.ReplaceAll(this.replaceTextbox.Text);
        }
        #endregion
    }
}